### PR TITLE
common/crc/aarch64: Added cpu feature pmull and make aarch64 specific…

### PR DESCRIPTION
**Pull request by aclamk**
… crc function to depend on it.

Signed-off-by: Adam Kupczyk <akupczyk@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
@Xenopathic does this change address https://tracker.ceph.com/issues/23464 ?

**Comment by aclamk** [comment]
@tchaikov @Xenopathic
 "pmull" instruction seems to be used only in crc function. So, it should fix 23464.

**Comment by tchaikov** [comment]
@aclamk could you link your change to https://tracker.ceph.com/issues/23464 in its commit message?

File: src/common/crc32c_aarch64.c [hunk 1]
```diff
@@ -96,6 +97,7 @@ uint32_t ceph_crc32c_aarch64(uint32_t crc, unsigned char const *buffer, unsigned
 
 	if (buffer) {
 #ifdef HAVE_ARMV8_CRYPTO
+	        if (ceph_arch_aarch64_pmull) {
```
Comment (tchaikov) [inline src/common/crc32c_aarch64.c]: i think we should instead check this variable in `ceph_choose_crc32()`. i.e.:
```c
#elif defined(__arm__) || defined(__aarch64__)
  if (ceph_arch_aarch64_crc32 && ceph_arch_aarch64_pmull) {
    return ceph_crc32c_aarch64;
  }
```

Comment (aclamk) [reply src/common/crc32c_aarch64.c]: @tchaikov No. The check for crc32 is necessary to be able to use ceph_crc32c_aarch64. CRC opcode is a must.
PMULL allows further improvement, but it is optional.

**Comment by Xenopathic** [comment]
I did try to compile Ceph, I really tried. But after running out of RAM several times and it taking way too long, I can't spend anymore time attempting to test this, sorry.

File: src/common/crc32c_aarch64.c (hunk 1 above)
Comment (tchaikov) [reply src/common/crc32c_aarch64.c]: oic. thanks!

**Comment by tchaikov** [comment]
@Xenopathic thanks all the same!

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-05-25_12:39:01-rados-wip-kefu-testing2-2018-05-25-1319-distro-basic-smithi/

**Pull request merged**